compiler/src/iree/compiler/Dialect/Flow/Transforms/test/hoist_encoding_ops.mlir

```diff
@@ -0,0 +1,88 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(util.func(iree-flow-hoist-encoding-ops))" --split-input-file %s | FileCheck %s
```

Comment:
I am going to add some more tests, but this one shows how the pass works

Reply:
this test is quite large and difficult to maintain - can you reduce it? if you want larger integration tests those can live in e2e/ but in general unit tests should be kept as small as possible and test as little as possible to cover the behavior